### PR TITLE
Feature/ee2 final nfcens

### PR DESCRIPTION
**Pull request by SamiraArdani-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
In this PR, the EE2-final changes were performed for EVS-NFCENS component.

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
To perform EVS-NFCENS:
prep STEP driver script: dev/drivers/scripts/prep/nfcens/jevs_nfcens_wave_grid2obs_prep.sh
stats STEP driver script: dev/drivers/scripts/stats/nfcens/jevs_nfcens_wave_grid2obs_stats.sh
plots STEP driver script: dev/drivers/scripts/plots/nfcens/jevs_nfcens_wave_grid2obs_plots.sh

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by PerryShafran-NOAA** [comment]
Hi, @SamiraArdani-NOAA - I am beginning work on your PR and have submitted the prep step.

**Comment by PerryShafran-NOAA** [comment]
The prep step is finished.  

Please take a look at the output file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr325test/EVS/dev/drivers/scripts/prep/nfcens/jevs_nfcens_wave_grid2obs_prep.o107801239

And the output here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/nfcens/wave.20231030/nfcens/grid2obs

Please pay extra attention to whether the EE2 items that you have added to the code are being followed.

**Comment by SamiraArdani-NOAA** [comment]
Hi @PerryShafran-NOAA ,
Prep STEP outputs look good. 
I checked the prep .o file and no error keyword was found.

**Comment by PerryShafran-NOAA** [comment]
Great!  I will move onto stats then.

Perry

**Comment by PerryShafran-NOAA** [comment]
@SamiraArdani-NOAA the stats job is finished.

See the .o file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr325test/EVS/dev/drivers/scripts/stats/nfcens/jevs_nfcens_wave_grid2obs_stats.o107805237

And the output is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/nfcens/_____.20231030

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA 
Wile the stat outputs look good, I see error messages in stats .o file: 
timeout: failed to run command â ecflow_clientâ : No such file or directory
/apps/ops/prod/nco/core/prod_util.v2.0.13/ush/err_exit: line 57: ECF_JOBOUT: parameter null or not set.
I am working on it.

**Comment by PerryShafran-NOAA** [comment]
@SamiraArdani-NOAA it looks like you just didn't have the correct syntax of the export err line.  It should be:

export err=$?; err_chk

You have in your exevs script:

`export err=\$?; err_chk`

Remove that slash and this should be fixed.  

Change line 104, but keep in line 105 a it is since you are echoing that to a file.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA 
I applied the changes and pushed them.

**Comment by PerryShafran-NOAA** [comment]
Hi, @SamiraArdani-NOAA - please be sure those changes are, indeed, pushed into this feature branch.  I don't see that any changes have been made.

**Comment by PerryShafran-NOAA** [comment]
OK, I have the changes now, and I am re-running.

Perry

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : Oops. now it is showing the changes.

**Comment by PerryShafran-NOAA** [comment]
Hi, @SamiraArdani-NOAA 

Looks like the stats job is finished.  Interestingly, the .o file is much smaller.  

Please look at the .o file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr325test/EVS/dev/drivers/scripts/stats/nfcens/jevs_nfcens_wave_grid2obs_stats.o107811513

And the stats output in the same place as before.  Let me know if the stats is still good, and all checks out with the.o file.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : I am checking stats now. will be back to you shortly.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : The stat outputs look good. The .o file look good with no error.

**Comment by PerryShafran-NOAA** [comment]
Thanks!  The plots job is underway.

**Comment by PerryShafran-NOAA** [comment]
@SamiraArdani-NOAA The plots job is complete.

Please see the .o file:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr325test/EVS/dev/drivers/scripts/plots/nfcens/jevs_nfcens_grid2obs_plots.o107818815

And here are the plot tarballs:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/plots/nfcens/wave.20231030

Please note that they would up in the noscrub directory because you still have COMOUT set to the noscrub directory.  That means that when they get back into the parallel, the plot tarballs will be in noscrub.  Be sure you have COMOUT set to the ptmp directory in your plots driver script.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : The output plots and the .o file look good. Thank you!

**Comment by PerryShafran-NOAA** [comment]
Super!  We'll need you to do a sync fork before we merge.  Let me do a quick code review and then we'll be good to go!

File: scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh [hunk 1]
```diff
@@ -78,13 +82,13 @@ else
...
+  echo '*** FATAL ERROR: NO NFCENS *.stat FILE *** '
   echo "             for ${VDATE} "
   echo '**************************************** '
   echo ' '
   echo "${MODELNAME}_${RUN} $VDATE $vhour : NFCENS *.stat file missing."
   [[ "$LOUD" = YES ]] && set -x
-  ./postmsg "$jlogfile" "FATAL ERROR : NO NFCENS *.stat file for ${VDATE}"
+  ./postmsg "$jlogfile" "FATAL ERROR: NO NFCENS *.stat file for ${VDATE}"
   err_exit "FATAL ERROR: Did not copy the NFCENS *.stat file for ${VDATE}"
   exit
```
Comment (PerryShafran-NOAA) [inline scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh:92]: @SamiraArdani-NOAA I don't think a missing stat file is supposed to cause a FATAL ERROR for plotting?  Maybe this should be just a WARNING.  Plots should be able to run with a few missing stat files.  

Also, I don't think we're supposed to use jlogfiles unless we have compiled code, which we don't have.  

@AliciaBentley-NOAA @ShelleyMelchior-NOAA for awareness.

Comment (PerryShafran-NOAA) [reply scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh:92]: FATAL ERROR is fine here.  I didn't realize that this block of code is for the case when there are zero stat files found.  You can leave this the way it is.

However, we do need you to remove all references to jlogfile.  Just remove the part that says 

`./postmsg "$jlogfile"`

from this line and we'll be OK.  Remove all references to jlogfile in your code.

**Comment by AliciaBentley-NOAA** [comment]
@SamiraArdani-NOAA In the following block of code, can you please change "file" to "files" and "FILE" to "FILES" if the fatal error only occurs when there are no .stat files?

```
else
  set -x
  echo ' '
  echo '**************************************** '
  echo '*** FATAL ERROR: NO NFCENS *.stat FILE *** '
  echo "             for ${VDATE} "
  echo '**************************************** '
  echo ' '
  echo "${MODELNAME}_${RUN} $VDATE $vhour : NFCENS *.stat file missing."
  [[ "$LOUD" = YES ]] && set -x
  ./postmsg "$jlogfile" "FATAL ERROR: NO NFCENS *.stat file for ${VDATE}"
  err_exit "FATAL ERROR: Did not copy the NFCENS *.stat file for ${VDATE}"
  exit
fi
```

File: scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh (hunk 1 above)
Comment (SamiraArdani-NOAA) [reply scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh:92]: @PerryShafran-NOAA : Ok, I can change the "FATAL" to "WARNING" and remove the jlogfile.

Comment (PerryShafran-NOAA) [reply scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh:92]: @SamiraArdani-NOAA No, don't change FATAL to warning.  I realized later that FATAL ERROR is actually appropriate here.

**Comment by AliciaBentley-NOAA** [comment]
@SamiraArdani-NOAA Looks like [Line 43 on the EE2 checklist](https://docs.google.com/spreadsheets/d/1nfF7PMsdylGtwyQy-vLZSUw04jAih9V0uthd0e4tvoo/edit?pli=1#gid=364442932) for NFCENS was set to YES when it should have been NO. Can you please go through NFCENS and remove all references to jlogfile? I've done a search of the EVS auth repo, and there are a few instances in NFCENS:

https://github.com/search?q=repo%3ANOAA-EMC%2FEVS+jlogfile&type=code

**Comment by SamiraArdani-NOAA** [comment]
@AliciaBentley-NOAA : ok, I will remove all references to jlogfile.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : I applied the requested changes.

**Comment by AliciaBentley-NOAA** [comment]
@SamiraArdani-NOAA This is from Shelley in a separate thread, so you wouldn't have seen it:

Please delete the entire line (`postmsg "$jlogfile" "$msg"`). postmsg is a prod_util utility that requires 2 arguments. It will not work w/ your proposed construct.

You can see more here:
vi /apps/ops/prod/nco/core/prod_util.v2.0.8/ush/postmsg

Best,
Shelley

**Comment by SamiraArdani-NOAA** [comment]
Got it.

File: scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply scripts/plots/nfcens/exevs_nfcens_wave_grid2obs_plots.sh:92]: @SamiraArdani-NOAA Please change back to FATAL ERROR.  I was wrong to suggest that it should be WARNING.  Clarify for me that this is a case where the code found zero stat files for plotting.  

Also, remove the .postmsg from the above as well, please.

Thanks!

**Comment by PerryShafran-NOAA** [comment]
Thank you @SamiraArdani-NOAA !  One more item, we'll need you to do a sync fork and then we are good to merge this code.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : I fixed the error message and removed the postmsg. Thank you!

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : It shows that the branch is up-to-date.

**Comment by PerryShafran-NOAA** [comment]
Ah.  Thanks!

**Review by PerryShafran-NOAA — APPROVED** [review]
Approved for merge.

**Comment by SamiraArdani-NOAA** [comment]
@PerryShafran-NOAA : Thank you very much.

**Pull request merged**